src/main/java/com/fvv/bookstore/controller/BookControllerImpl.java

```diff
@@ -32,9 +32,12 @@ public BookControllerImpl() {
 	/**
 	 * {@inheritDoc}
 	 */
-	public boolean addBook(final Book book) throws ControllerException {
+	public boolean addBook(final Book book) throws BookValidationException, ControllerException {
 		try {
+			this.validateBook(book);
 			return this.bookDao.addBook(book);
+		} catch (BookValidationException e) {
```

Comment:
you can simply remove this catch here since you're re-throwing an existent one!

Reply:
ok